main.tf

```diff
@@ -276,3 +276,27 @@ resource "aws_spot_instance_request" "this" {
   tags        = merge({ "Name" = var.name }, var.tags)
   volume_tags = var.enable_volume_tags ? merge({ "Name" = var.name }, var.volume_tags) : null
 }
+
+locals {
+  external_ebs = { for ebs in var.ebs_block_device : ebs.device_name => ebs }
+}
+
+resource "aws_ebs_volume" "this" {
+  for_each = var.create && !var.override_ebs_mapping ? local.external_ebs : {}
+
+  encrypted         = lookup(each.value, "encrypted", null)
+  iops              = lookup(each.value, "iops", null)
+  kms_key_id        = lookup(each.value, "kms_key_id", null)
+  snapshot_id       = lookup(each.value, "snapshot_id", null)
+  size              = lookup(each.value, "volume_size", null)
+  type              = lookup(each.value, "volume_type", null)
+  throughput        = lookup(each.value, "throughput", null)
+  availability_zone = lookup(each.value, "availability_zone", try(aws_instance.this[0].availability_zone, aws_spot_instance_request.this[0].availability_zone))
```

Comment:
User can selectively assign which AZ to be created in. If not, it will follow where the instance is located